OpenChange/MAPIStoreTasksFolder.m

```diff
@@ -116,18 +133,64 @@ - (uint32_t) exchangeRightsForRoles: (NSArray *) roles
            && [roles containsObject: SOGoCalendarRole_PrivateViewer]
            && [roles containsObject: SOGoCalendarRole_ConfidentialViewer])
     rights |= RightsReadItems;
+
+  if ([roles containsObject: MAPIStoreRightEditOwn])
+    rights |= RightsEditOwn;
+  if ([roles containsObject: MAPIStoreRightDeleteOwn])
+    rights |= RightsDeleteOwn;
+
+  if ([roles containsObject: MAPIStoreRightFolderOwner])
+    rights |= RightsFolderOwner | RoleNone;
```

Comment:
I will move the Folder Visible above to avoid setting FolderVisible when only FolderContact was set as the user expects. This was nicely suggested from @blaxter